langchain/memory/chat_message_histories/redis.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+from typing import List, Optional
+
+from langchain.schema import (
+    AIMessage,
+    BaseChatMessageHistory,
+    BaseMessage,
+    HumanMessage,
+    _message_to_dict,
+    messages_from_dict,
+    messages_to_dict,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class RedisChatMessageHistory(BaseChatMessageHistory):
+    def __init__(
+        self,
+        session_id: str,
+        url: str = "redis://localhost:6379/0",
+        key_prefix: str = "message_store:",
+        ttl: Optional[int] = None,
+    ):
+        try:
+            import redis
+        except ImportError:
+            raise ValueError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
+            )
+
+        try:
+            self.redis_client = redis.Redis.from_url(url=url)
+        except redis.exceptions.ConnectionError as error:
+            logger.error(error)
+
+        self.session_id = session_id
+        self.key_prefix = key_prefix
+        self.ttl = ttl
+
+    def get_key(self, session_id: str) -> str:
+        """Construct the record key to use"""
+        return self.key_prefix + session_id
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
```

Comment:
So my recommendation would be doing this with a Redis List data type. OR Redis JSON. But Lists are built -in and don't need any exyta modules. Instead of storing things in a json dumped blob, you can stor messages in a list.

https://redis.io/docs/data-types/lists/
https://redis-py.readthedocs.io/en/stable/commands.html#redis.commands.core.CoreCommands.lpush

Reply:
This will also make appening to the list much easier, and done on the db side, not client side

Reply:
@tylerhutcherson i took a stab at updating - lmk wht you think!

Reply:
@hwchase17,
What happened to the TTL ([see](https://github.com/hwchase17/langchain/pull/2122/commits/dd72715410fca61de1e7bcebe0b4551ddba04ceb#diff-7ef42aa53f31305f6c6ea1c5e94084f5164f274e6de1575322e6e61dad018138L72))?
I am fine changing to using a Redis List data type, but it seems we have also dropped the TTL option with this change.

I don't think it is a good idea not to have a TTL, as that means that these records would be kept in Redis forever with no expiration option.

Outside it probably not being a good idea to accumulate data forever in Redis, also most data privacy laws disallow that.

Reply:
This is true. Though managing the size of that list could be simple, and even configurable. One mode of operation could be maintaining the list at a **fixed size.** That combined with overall conversation TTL should be sufficient.